airbyte-integrations/connectors/source-facebook-marketing/integration_tests/test_streams.py

```diff
@@ -32,16 +31,22 @@ def state_with_include_deleted_fixture(state):
 
 
 @pytest.fixture(scope="session", name="configured_catalog")
-def configured_catalog_fixture():
-    return ConfiguredAirbyteCatalog.parse_file("integration_tests/configured_catalog_without_insights.json")
+def configured_catalog_fixture(config) -> ConfiguredAirbyteCatalog:
+    catalog = SourceFacebookMarketing().discover(logger=logging.getLogger("airbyte"), config=config)
+    streams = [
+        ConfiguredAirbyteStream(stream=stream, sync_mode=stream.supported_sync_modes[0], destination_sync_mode=DestinationSyncMode.append)
+        for stream in catalog.streams
+    ]
+    return ConfiguredAirbyteCatalog(streams=streams)
 
 
 class TestFacebookMarketingSource:
     @pytest.mark.parametrize(
         "stream_name, deleted_id", [("ads", "23846756820320398"), ("campaigns", "23846541919710398"), ("ad_sets", "23846541706990398")]
     )
     def test_streams_with_include_deleted(self, stream_name, deleted_id, config_with_include_deleted, configured_catalog):
-        catalog = self.slice_catalog(configured_catalog, {stream_name})
+        catalog = self._slice_catalog(configured_catalog, {stream_name})
+        catalog.streams[0].sync_mode = SyncMode.incremental
```

Comment:
could you add some context for why we're choosing the 0th stream to be incremental? it's not clear why this is necessary

Reply:
this is necessary because it is a list of a single stream, will add comment